### PR TITLE
cmd/ed25519: ed25519 cmd

### PR DESCRIPTION
**Pull request by bobg**
As described in https://gist.github.com/kr/2977d492d6afc428e31f49dee13dc3a7

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

**Comment by kr** [comment]
:eyes:

File: cmd/ed25519/ed25519.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+func main() {
+	if len(os.Args) < 2 {
+		usage()
+	}
+	switch os.Args[1] {
+	case "gen":
+		_, prv, err := ed25519.GenerateKey(nil)
+		must(err)
+		os.Stdout.Write([]byte(prv))
```
Comment (kr) [inline cmd/ed25519/ed25519.go]: nit: the `[]byte(...)` conversion is unnecessary

File: cmd/ed25519/ed25519.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			output = "BAD"
+		}
+		if silent {
+			w = os.Stderr
+		} else {
+			w = os.Stdout
+		}
+		fmt.Fprintln(w, output)
+		if silent && !ok {
+			os.Exit(1)
```
Comment (kr) [inline cmd/ed25519/ed25519.go]: Any reason not to set the exit status in the not-silent mode as well?

File: cmd/ed25519/ed25519.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			output string
+			w      io.Writer
+		)
+		if ok {
+			output = "OK"
+		} else {
+			output = "BAD"
+		}
+		if silent {
+			w = os.Stderr
```
Comment (kr) [inline cmd/ed25519/ed25519.go]: I feel like maybe silent mode should not print anything (to either output stream). I can imagine trying to use this in a shell script and having to provide both `-s` and also redirect stderr to /dev/null. What do you think?

Comment (bobg) [inline cmd/ed25519/ed25519.go]: I agree, but the gist says otherwise. Switched to be silent in silent mode.

File: cmd/ed25519/ed25519.go (hunk 2 above)
Comment (bobg) [inline cmd/ed25519/ed25519.go]: Yeah, in non-silent mode it's not failure successfully to detect a bad signature. Using this in a script and parsing its output, you want it to exit with status 0 to not affect the flow of the script.

**Comment by bobg** [comment]
PTAL

File: cmd/ed25519/ed25519.go (hunk 2 above)
Comment (bobg) [inline cmd/ed25519/ed25519.go]: Shrug, OK.

**Comment by kr** [comment]
LGTM

**Comment by bobg** [comment]
rebased

**Comment by bobg** [comment]
rerebased

**Pull request merged**